File: src/ragapp/backend/models/tools/query_engine.py
from typing import ClassVar, Literal, Optional

from pydantic import BaseModel


class QueryEngineTool(BaseModel):
    config_id: ClassVar[str] = "query_engine"
    name: Literal["query_engine"] = "query_engine"
    tool_type: Literal["local"] = "local"
    label: Literal["Query Engine"] = "Query Engine"
    custom_prompt: Optional[str] = None
    description: str = "Query the database"
    config: dict = {}
    enabled: bool = False

File: src/ragapp/admin-ui/client/tools/query_engine.ts
import { z } from "zod";

export const QueryEngineToolConfig = z.object({
  name: z.literal("query_engine"),
  label: z.string(),
  description: z.string(),
  enabled: z.boolean().nullable().optional(),
  config: z.object({}).nullable().optional(),
  priority: z.number(),
});

export type QueryEngineToolConfigType = z.infer<typeof QueryEngineToolConfig>;

export const DEFAULT_QUERY_ENGINE_TOOL_CONFIG: QueryEngineToolConfigType = {
  name: "query_engine",
  label: "Use Knowledge",
  description: "Query information from your knowledge base",
  config: {},
  enabled: false,
  priority: 1, // Assign a priority value
};
